crates/fluvio-controlplane-metadata/src/managed_connector/k8/spec.rs

```diff
@@ -0,0 +1,78 @@
+//!
+//! # SPU Spec
+//!
+//! Interface to the SPU metadata spec in K8 key value store
+//!
+use serde::Deserialize;
+use serde::Serialize;
+
+use super::super::ManagedConnectorStatus;
+use crate::k8_types::{Spec, Crd, DefaultHeader};
+
+use crd::MANAGED_CONNECTOR_API;
+mod crd {
+
+    use crate::k8_types::{Crd, CrdNames, GROUP, V1};
+
+    pub const MANAGED_CONNECTOR_API: Crd = Crd {
+        group: GROUP,
+        version: V1,
+        names: CrdNames {
+            kind: "ManagedConnector",
+            plural: "managedconnectors",
+            singular: "managedconnector",
+        },
+    };
+}
+
+impl Spec for K8ManagedConnectorSpec {
+    type Status = ManagedConnectorStatus;
+    type Header = DefaultHeader;
+    fn metadata() -> &'static Crd {
+        &MANAGED_CONNECTOR_API
+    }
+}
+use super::ManagedConnectorConfig;
+
+#[derive(Deserialize, Serialize, Default, Debug, Clone)]
+#[serde(rename_all = "camelCase", default)]
+pub struct K8ManagedConnectorSpec {
+    pub name: String,
+    pub config: ManagedConnectorConfig,
```

Comment:
Is there reason why config is separate from name?

Reply:
I think this should be flattened